src/Microsoft.AspNetCore.Mvc.TagHelpers/LinkTagHelper.cs

```diff
@@ -341,7 +358,59 @@ private void BuildFallbackBlock(TagHelperContent builder)
                 .AppendHtml("\",");
 
             AppendFallbackHrefs(builder, fallbackHrefs);
-            builder.AppendHtml("</script>");
+
+            builder.AppendHtml(", \"");
+
+            // Perf: Avoid allocating enumerator
+            for (var i = 0; i < attributes.Count; i++)
+            {
+                var attribute = attributes[i];
+                if (string.Equals(attribute.Name, HrefAttributeName, StringComparison.OrdinalIgnoreCase))
+                {
+                    continue;
+                }
+
+                attribute.WriteTo(StringWriter, HtmlEncoder);
+                StringWriter.Write(' ');
+            }
+
+            var stringBuilder = StringWriter.GetStringBuilder();
+            var scriptTags = stringBuilder.ToString();
+            stringBuilder.Clear();
+            var encodedScriptTags = JavaScriptEncoder.Encode(scriptTags);
+            builder.AppendHtml(encodedScriptTags);
+
+            builder.AppendHtml("\");</script>");
+        }
+
+        private bool HasRelStylesheet(TagHelperAttributeList attributes)
+        {
+            TagHelperAttribute relAttribute;
+            if (!attributes.TryGetAttribute(RelAttributeName, out relAttribute) ||
+                relAttribute.Value == null)
+            {
+                return false;
+            }
+
+            var attributeValue = relAttribute.Value;
+            var contentValue = attributeValue as IHtmlContent;
+            var stringValue = attributeValue as string;
+            if (contentValue != null)
+            {
+                contentValue.WriteTo(StringWriter, HtmlEncoder);
+                stringValue = StringWriter.ToString();
+
+                // Reset writer
+                StringWriter.GetStringBuilder().Clear();
+            }
+            else if (stringValue == null)
+            {
+                stringValue = attributeValue.ToString();
```

Comment:
> Never make the assumption that ToString() returns a useful value. Only a few IHtmlContent implementations even override ToString(); they are classes w/ a WriteTo() method and that's it.

We're not, in the first if we're seeing if it's an IHtmlContent. If so, calling `WriteTo`. This `ToString()` right here is only called when the value is not a string and not an `IHtmlContent`


> I thought the decision was to ignore everything beyond a literal rel='stylesheet' or rel="stylesheet"

We considered using the CSS attribute selectors (which expect purely literal content) but that'd be a decently large behavior change. Therefore, I'm just ensuring I can read the `attribute.Value` no matter what it might be. Doesn't hurt to be fool proof here. It's `object` after all.